ethereum/api/src/test/java/org/hyperledger/besu/ethereum/api/jsonrpc/internal/privacy/methods/priv/PrivCallTest.java

```diff
@@ -0,0 +1,196 @@
+    final CallParameter callParameter = new JsonCallParameter("0x0", null, "0x0", "0x0", "0x0", "");
+    final JsonRpcRequestContext request = ethCallRequest(privacyGroupId, callParameter, "latest");
+
+    final Throwable thrown = catchThrowable(() -> method.response(request));
+
+    assertThat(thrown)
+        .isInstanceOf(InvalidJsonRpcParameters.class)
+        .hasNoCause()
+        .hasMessage("Missing \"to\" field in call arguments");
+  }
+
+  @Test
+  public void shouldReturnNullWhenProcessorReturnsEmpty() {
+    final JsonRpcRequestContext request = ethCallRequest(privacyGroupId, callParameter(), "latest");
+    final JsonRpcResponse expectedResponse = new JsonRpcSuccessResponse(null, null);
+
+    when(privateTransactionSimulator.process(any(), any(), any(), anyLong()))
+        .thenReturn(Optional.empty());
+
+    final JsonRpcResponse response = method.response(request);
+
+    assertThat(response).isEqualToComparingFieldByField(expectedResponse);
+    verify(privateTransactionSimulator).process(any(), any(), any(), anyLong());
+  }
+
+  @Test
+  public void shouldAcceptRequestWhenMissingOptionalFields() {
+    final CallParameter callParameter = new JsonCallParameter(null, "0x0", null, null, null, null);
+    final JsonRpcRequestContext request = ethCallRequest(privacyGroupId, callParameter, "latest");
+    final JsonRpcResponse expectedResponse =
+        new JsonRpcSuccessResponse(null, Bytes.of().toString());
+
+    mockTransactionProcessorSuccessResult(Bytes.of());
+
+    final JsonRpcResponse response = method.response(request);
+
+    assertThat(response).isEqualToComparingFieldByFieldRecursively(expectedResponse);
+    verify(privateTransactionSimulator).process(any(), any(), eq(callParameter), anyLong());
+  }
+
+  @Test
+  public void shouldReturnExecutionResultWhenExecutionIsSuccessful() {
+    final JsonRpcRequestContext request = ethCallRequest(privacyGroupId, callParameter(), "latest");
+    final JsonRpcResponse expectedResponse =
+        new JsonRpcSuccessResponse(null, Bytes.of(1).toString());
+    mockTransactionProcessorSuccessResult(Bytes.of(1));
+
+    final JsonRpcResponse response = method.response(request);
+
+    assertThat(response).isEqualToComparingFieldByFieldRecursively(expectedResponse);
+    verify(privateTransactionSimulator).process(any(), any(), eq(callParameter()), anyLong());
+  }
+
+  @Test
+  public void shouldUseCorrectBlockNumberWhenLatest() {
+    final JsonRpcRequestContext request = ethCallRequest(privacyGroupId, callParameter(), "latest");
+    when(blockchainQueries.headBlockNumber()).thenReturn(11L);
+    when(privateTransactionSimulator.process(any(), any(), any(), anyLong()))
+        .thenReturn(Optional.empty());
+
+    method.response(request);
+
+    verify(privateTransactionSimulator).process(any(), any(), any(), eq(11L));
+  }
+
+  @Test
+  public void shouldUseCorrectBlockNumberWhenEarliest() {
```

Comment:
If we use anything but "latest", the method is supposed to fail, isn't it? This test seems to be testing the wrong thing (or it is misleading)

Reply:
This is testing the PrivateTransactionSimulator, and the test whether we are using the latest block is in the PrivCall class.